bpf/kmesh/sockops_workload.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <linux/bpf.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_endian.h>
+#include "bpf_log.h"
+
+#define AF_INET  (2)
+#define AF_INET6 (10)
+
+#define RINGBUF_SIZE (1 << 12)
+
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, RINGBUF_SIZE);
+} map_of_mgrt SEC(".maps");
+
+enum family_type {
+    IPV4,
+    IPV6,
+};
+
+struct ringbuf_msg_type {
+    __u32 type; 
+    struct bpf_sock_tuple tuple;
+};
+
+SEC("sockops")
+int socket_migration(struct bpf_sock_ops *skops)
+{
+    struct ringbuf_msg_type *msg;
+
+    switch (skops->op) {
+        case BPF_SOCK_OPS_PASSIVE_ESTABLISHED_CB:           
```

Comment:
I'd rather optimizing by check the workload map, and only if the dst address has correlated workload, we will do the following, othwrsie just break

Reply:
An interface ```dst_is_managed()``` is reserved in ```sockops_workload.c``` to fix this issue